doc/src/mysql.rst

```diff
@@ -0,0 +1,27 @@
+.. _nixos-mysql:
+
+MySQL
+======
+
+Managed instance of the `MySQL`_ database server.
```

Comment:
the postgresql role links to postgresql.org here but I couldn't find a good link to put here, as they're all implementation specific